service/src/test/kotlin/no/nav/su/se/bakover/service/søknadsbehandling/SøknadsbehandlingServiceIverksettTest.kt

```diff
@@ -314,115 +196,82 @@ internal class SøknadsbehandlingServiceIverksettTest {
-                    attestering = attestering,
-                ),
-            ) shouldBe expectedAvslag.right()
-
-            verify(it.søknadsbehandlingRepo).hent(avslagTilAttestering.id)
-            verify(it.brevService).lagDokument(argThat { it shouldBe beOfType<Avslagsvedtak.AvslagBeregning>() })
-            verify(it.søknadsbehandlingRepo).lagre(eq(expectedAvslag), anyOrNull())
-            verify(it.vedtakRepo).lagre(argThat { it is Avslagsvedtak.AvslagBeregning }, anyOrNull())
-            verify(it.brevService).lagreDokument(
-                argThat {
-                    it.metadata.sakId shouldBe avslagTilAttestering.sakId
-                    it.metadata.vedtakId shouldNotBe null
-                    it.metadata.bestillBrev shouldBe true
-                },
-                anyOrNull()
-            )
-            verify(it.behandlingMetrics).incrementAvslåttCounter(BehandlingMetrics.AvslåttHandlinger.PERSISTERT)
-            verify(it.ferdigstillVedtakService).lukkOppgaveMedBruker(any())
-            verify(it.observer).handle(
-                argThat {
-                    it shouldBe Event.Statistikk.SøknadsbehandlingStatistikk.SøknadsbehandlingIverksatt(
-                        expectedAvslag,
-                    )
-                },
-            )
-            it.verifyNoMoreInteractions()
-        }
+        )
+
+        serviceAndMocks.søknadsbehandlingService.iverksett(
+            SøknadsbehandlingService.IverksettRequest(
+                behandlingId = avslagTilAttestering.id,
+                attestering = attestering,
+            ),
+        ) shouldBe expectedAvslag.right()
+
+        verify(serviceAndMocks.søknadsbehandlingRepo).hent(avslagTilAttestering.id)
+        verify(serviceAndMocks.brevService).lagDokument(argThat { it shouldBe beOfType<Avslagsvedtak.AvslagBeregning>() })
+        verify(serviceAndMocks.søknadsbehandlingRepo).lagre(eq(expectedAvslag), anyOrNull())
+        verify(serviceAndMocks.vedtakRepo).lagre(argThat { it is Avslagsvedtak.AvslagBeregning }, anyOrNull())
+        verify(serviceAndMocks.brevService).lagreDokument(
+            argThat {
+                it.metadata.sakId shouldBe avslagTilAttestering.sakId
+                it.metadata.vedtakId shouldNotBe null
+                it.metadata.bestillBrev shouldBe true
+            },
+            anyOrNull()
+        )
+        verify(serviceAndMocks.behandlingMetrics).incrementAvslåttCounter(BehandlingMetrics.AvslåttHandlinger.PERSISTERT)
+        verify(serviceAndMocks.ferdigstillVedtakService).lukkOppgaveMedBruker(any())
+        verify(serviceAndMocks.observer).handle(
+            argThat {
+                it shouldBe Event.Statistikk.SøknadsbehandlingStatistikk.SøknadsbehandlingIverksatt(expectedAvslag)
+            },
+        )
+        serviceAndMocks.verifyNoMoreInteractions()
     }
 
     @Test
     fun `iverksett behandling attesterer og saksbehandler kan ikke være samme person`() {
```

Comment:
```suggestion
    fun `attestant og saksbehandler kan ikke være samme person`() {
```

Denne bør kanskje flyttes opp sammen med de andre feil-scenariene.

Reply:
Ja, helt enig, men tenkte det var lettere å se diffen på testen om jeg lot den ligge